TYPO3.Media/Classes/TYPO3/Media/Domain/Service/AssetService.php

```diff
@@ -32,71 +35,47 @@ class AssetService
     protected $thumbnailService;
 
     /**
+     * @Flow\Inject
+     * @var UriBuilder
+     */
+    protected $uriBuilder;
+
+    /**
      * Calculates the dimensions of the thumbnail to be generated and returns the thumbnail URI.
      * In case of Images this is a thumbnail of the image, in case of other assets an icon representation.
      *
      * @param AssetInterface $asset
      * @param ThumbnailConfiguration $configuration
+     * @param boolean $async Return asynchronous image URI in case the requested image does not exist already
+     * @param ActionRequest $request Request argument must be provided for asynchronous thumbnails
      * @return array with keys "width", "height" and "src"
+     * @throws AssetServiceException
      */
-    public function getThumbnailUriAndSizeForAsset(AssetInterface $asset, ThumbnailConfiguration $configuration)
+    public function getThumbnailUriAndSizeForAsset(AssetInterface $asset, ThumbnailConfiguration $configuration, $async = false, ActionRequest $request = null)
```

Comment:
See previous comment about the ThumbnailConfiguration and the async property, not sure if the ActionRequest should go to the ThumbnailConfiguration